doc/limit/SONiC Limit user session and memory.md

```diff
@@ -0,0 +1,325 @@
+| ------------ | --------------------------------------------- | ----------------------------------------------- | ------------------------------ | --------------------------- | ---------------- | --------- | ------ |
+| rlimit       | PAM plugin                                    |                                                 | Only per-procress control.     | per-user per-group control. | NO               | NO        | NO     |
+| cgroup-tools | daemon manage/migrate user process to cgroups | Cross procress control. Support user group.     | Not support domain user.       | Yes                         | NO               | Yes       |        |
+| systemd      | Manage cgroups with systemd user.slices       | Cross procress control. Support slice template. | Not support per-group control. | Yes                         | Yes              | No        | Yes    |
+
+- rlimit only support per-procress control.
+- cgroup-tools not support domain user, this is because SONiC will create new user when domain user login, but cgroup-tools only support existed user.
+- systemd have good support to domain user, but not support per-group limit.
+
+
+
+## 3.3 Default memory limitation Implementation
+- Max number of logins by memory size:
+  - Max number of logins = memory size * factor / max memory per-user.
+- Default factor by OS version, device type and vendor, which is based on history data of SONiC memory utilization:
+  - Celestica:
+    - M0: 0.4
+    - T0 & T1: 0.6
+  - Mellanox & Nexus:
+    - T0 & T1: 0.6
+  - Nokia:
+    - M0: 0.6
+  - Arista:
+    - T0: 0.7
+    - T1: 0.6
+  - Dell & Firce10
+    - T0 & T1: 0.7
+  - For all other device, default factor is 0.4
+- Max memory per-user is hardcode config. Default value is 200 MB, because 'show' command will take 70 MB memory, and we plan support user will run 3 commands concurrently.
+  - If user want run a script/command which take more than 200MB memory and been blocked by this feature, user can modify the per-user limit with the config command.
+- For customer, they may have pipelines to initialize device configuration, because this feature add new commands, the pipeline may need update. The default limitation is designed to cover most case to minimize the pipeline change.
+
+## 3.4 ConfigDB Schema
+ - Limit setting table.
+```
+; Key
+limit_key              = 1*32VCHAR          ; setting name, format is resource type + limit scope + limit name
+; Attributes
+resource_type          = LIST(1*32VCHAR)   ; Limit resource type, now only support (login, memory)
+scope                  = LIST(1*32VCHAR)   ; Limit scope, now only support (global, group, user)
+value                  = Number  ; limit value, for login this is max login session count, for memory this is memory side in byte.
+```
```

Comment:
can you also add yang model?

Reply:
Fixed, add yang model and link to design document.